src/app/page.tsx

```diff
@@ -33,15 +34,48 @@ const transactionElement = (transaction: Transaction) => {
 
 export default function Page() {
   const { data, error, isLoading } = useSWR("/api/transactions", fetcher);
+  const [dates, setDates] = useState([new Date(0), new Date()]);
+
+  const dateFilter = (transaction) => {
+    if (
+      new Date(transaction.timestamp) > dates[0] &&
+      new Date(transaction.timestamp) < dates[1]
+    ) {
+      return transaction;
+    }
+  };
+
+  const monthsAgo = (n) => {
+    const date = new Date();
+    const month = date.getMonth();
+    date.setMonth(month - n);
+    return date;
+  };
 
   if (error) return <div>Failed to load</div>;
   if (isLoading) return <div>Loading...</div>;
 
   return (
     <div className="max-w-md mx-auto py-12 space-y-8">
       <h1 className="text-4xl">Transactions</h1>
+      <div className="flex gap-4">
+        <button
+          type="button"
+          className="py-2 px-3 border border-zinc-500 rounded"
+          onClick={() => setDates([monthsAgo(1), new Date()])}
+        >
+          This month
+        </button>
+        <button
+          type="button"
+          className="py-2 px-3 border border-zinc-500 rounded"
+          onClick={() => setDates([monthsAgo(3), new Date()])}
+        >
+          Last 3 months
+        </button>
+      </div>
       <ul className="space-y-2 pb-8">
-        {data.transactions.map(transactionElement)}
+        {data.transactions.filter(dateFilter).map(transactionElement)}
```

Comment:
Is reducing unnecessary load on the DB a requirement? If so, this doesn't fulfil that as we're still fetching all the transactions from the DB and only filtering on the client side. However, this is still an improvement over what we had before, so I'm fine with this if we're happy with it 👍

It would be good to look at filtering at the DB level with date parameters going forward though.